src/web/types/form-validator.ts

```diff
@@ -33,4 +33,35 @@ export enum FormValidator {
    * Max length for the 'E-mail Address` field.
    */
   EMAIL_MAX_LENGTH = ApiConst.EMAIL_MAX_LENGTH,
+
+  /**
+   * Regex used to verify emails in the back-end.
+   */
+  EMAIL_REGEX = ApiStringConst.EMAIL_REGEX,
+
+  /**
+   * Regex used to verify names.
+   *
+   * Based on back-end's `FieldValidator.REGEX_NAME`.
+   * The back-end regex is not converted to use here as the pattern syntax is not accepted in JS.
+   */
+  NAME_REGEX = '^[a-zA-Z0-9][^|%]*$',
```

Comment:
My concern with this is that the back-end regex and the front-end regex may go out of sync; I think it would probably be better to use the same regex in the front-end as in the back-end. Still, if the one in the back-end cannot be converted then I guess there is nothing much that we can do (that I know of).

One option would be to use your front-end regex in the back-end, but changing the one in the back-end is probably out of our scope (perhaps this can be done after `account-request-form` is merged into `master`). Another option might be to not check this in the front-end at all and just leave this for the back-end to check, but the feedback would probably take longer (requires requests to the server).

In any case, it's probably unlikely that this will go out of sync with the regex in the back-end because it seems it doesn't really change often. For me, I can accept this for now, but let's see if other reviewers want to resolve this and have ideas on how to do so.

Reply:
My other concern with leaving it for the back-end to check (besides the extra requests to the server) is that the display of errors in the front-end would be less clear if there are validation issues for multiple fields. Since the server response is not tied to a particular form field, the error(s) can only be displayed all at once at the bottom of the form and not near the relevant field(s).

I will add a comment to highlight the existence of the separate regexes.

Reply:
Would it be better to have the front-end regex in the back-end `FieldValidator` (alongside the existing back-end one) and get converted with the other API constants instead? This would reduce the likelihood of one being changed without noticing the other. However, it could be confusing as the front-end regex isn't actually used in any of the back-end code.

Reply:
I agree putting front-end regex in the back-end might be confusing since it is a pure front-end check. Maybe leave it in the front-end code?